File: src/main/java/ru/melonhell/bulletphysics/bullet/collision/space/cache/fluid/impl/EmptyFluidCacheFactory.java
package ru.melonhell.bulletphysics.bullet.collision.space.cache.fluid.impl;

import org.springframework.stereotype.Component;
import ru.melonhell.bulletphysics.bullet.collision.space.MinecraftSpace;
import ru.melonhell.bulletphysics.bullet.collision.space.cache.fluid.FluidCacheFactory;

@Component
public class EmptyFluidCacheFactory implements FluidCacheFactory {
    @Override
    public EmptyFluidCache create(MinecraftSpace space) {
        return new EmptyFluidCache();
    }
}

File: src/main/java/ru/melonhell/bulletphysics/bullet/collision/space/cache/fluid/impl/EmptyFluidCacheFactory.kt
package ru.melonhell.bulletphysics.bullet.collision.space.cache.fluid.impl

import org.springframework.stereotype.Component
import ru.melonhell.bulletphysics.bullet.collision.space.MinecraftSpace
import ru.melonhell.bulletphysics.bullet.collision.space.cache.fluid.FluidCacheFactory

@Component
class EmptyFluidCacheFactory : FluidCacheFactory {
    override fun create(space: MinecraftSpace): EmptyFluidCache = EmptyFluidCache()
}